File: testprojects/master_scala_mainone/pipelines/SCALA_SG_SRC/code/src/main/scala/graph/all_types/FlattenSchema_1.scala
package graph.all_types

import io.prophecy.libs._
import config.ConfigStore._
import config.Context
import udfs.UDFs._
import udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object FlattenSchema_1 {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.withColumn("c_array--long", explode_outer(col("c_array--long")))
      .withColumn("c_array -- decimal",
                  explode_outer(col("c_array -- decimal"))
      )
      .withColumn("c_array-int  _ int",
                  explode_outer(col("c_array-int  _ int"))
      )
      .withColumn(
        "c_struct -- _  -c_array_int - of a struct ",
        explode_outer(col("c_struct -- _  .c_array_int - of a struct "))
      )
      .select(
        col("c_array--long"),
        col("c_array -- decimal"),
        col("c_struct -- _  .c_array_int - of a struct ")
          .as("c_struct -- _  -c_array_int - of a struct "),
        col("c_struct -- _  .c_timestamp - of a struct-")
          .as("c_struct -- _  -c_timestamp - of a struct-"),
        col("c_struct -- _  .c_string - of a struct -- _")
          .as("c_struct -- _  -c_string - of a struct -- _"),
        col("c_array-int  _ int")
      )

}

File: testprojects/python_main/pipelines/PYTHON_DEP_MGMT_ALL/code/com/main1/pythondepmanagement_1/graph/FlattenSchema_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from com.main1.pythondepmanagement_1.config.ConfigStore import *
from com.main1.pythondepmanagement_1.udfs.UDFs import *

def FlattenSchema_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn("c_array--long", explode_outer("c_array--long"))\
        .withColumn("c_array-int  _ int", explode_outer("c_array-int  _ int"))\
        .withColumn("-- c_array_timestamp -- ", explode_outer("-- c_array_timestamp -- "))\
        .withColumn("c_array -- float", explode_outer("c_array -- float"))\
        .withColumn("c_struct -- _  -c_array_int - of a struct ", explode_outer("c_struct -- _  .c_array_int - of a struct "))\
        .select(col("c_struct -- _  .c_string - of a struct -- _").alias("c_struct -- _  -c_string - of a struct -- _"), col("c_struct -- _  .c_array_int - of a struct ").alias("c_struct -- _  -c_array_int - of a struct "), col("c_array--long"), col("c_array-int  _ int"), col("-- c_array_timestamp -- "), col("c_array -- float"))
